### DISCUSSION TITLE
why filament doesn't use opencolorio to implement color grading

### DISCUSSION THREAD

**Discussion by dandingol03** [google/filament#6986] [General]
Hi, i met a problem recently. The model rendered in `blender` has huge chromatic compared to `filament` .
 &nbsp; &nbsp;After a while, i finally find the reason why there is a lot chromatic when i use the same model and same hdr to provide environment indirect-light. 
 &nbsp; &nbsp;The reason is blender use `opencolorio` to implement color management, and i found the `unreal` engine also use it to do so. So why `filament` doesn't use `opencolorio`? This difference makes the artist must do the same work to make model rendering right after they have do the same stuff in `blender`

**Comment by romainguy** [comment]
OCIO is a large and complex system that is not well suited for Filament. More importantly, OCIO itself doesn't solve what you are describing. The difference you are observing is most likely due to a difference in tone mapping between Blender and Filament. Filament offers several tone mapping algorithms, including the ACES reference, that can help you get closer to Blender (Blender now uses its "Filmic" by default).

**Reply by dandingol03** [reply]
&nbsp; **No**, i have choose the `tonemapping` to `filmic` in filament.  However there is still difference between them. Here is some screenshot:
![image](https://github.com/google/filament/assets/13013579/fb1e9790-6d5a-48c5-9aa2-15e99520e6e5)
![image](https://github.com/google/filament/assets/13013579/4744c1ab-ef2d-4e4d-bc2e-bfce7350c4c0)

&nbsp; The left picture is the rendered result in `blend` and the right picture is belong to `filament`. You can see the right picture is more saturated when the model is placed in same indirect light environment.

**Reply by dandingol03** [reply]
If i baked the lut in opencolorio in advance which has the same ocio config with blender,  can i fix chromatic difference by sampling the lut in colorgrading.mat?

**Reply by romainguy** [reply]
Filmic is a generic term it doesn't mean it's the same tone mapper.

And setting aside saturation, it looks like the lighting is different, or the materials are different (less rough with filament)

**Reply by dandingol03** [reply]
&nbsp;&nbsp;I agree your first opinion, the implementation must be different. I think the biggest reason which make chromatic exist is the `filmic` implementation. Here is some picture for contrast:
![image](https://github.com/google/filament/assets/13013579/8e44b187-1fd5-4062-ae82-50af9de17798)
&nbsp;&nbsp;The left part is the render result in `filament`, and the other is rendered in `blender`. The most obvious chromatic is the curtain's color, and the curtain's is part of the skybox. Because the rendering result of skybox donot take light result into account, so different implementation of `filmic` tonemapper have most impact on chromatic problem.
The most i wanna know is how to fix this chromatic  problem. How if i replace lut texture in `colorgrading.mat` by the lut which is generated by `blender`.
&nbsp;&nbsp;Thanks
